exasol_transformers_extension/udfs/models/token_classification_udf.py

```diff
@@ -0,0 +1,199 @@
+                result_df_list.append(pred_df)
+
+        result_df = pd.concat(result_df_list)
+        return result_df
+
+    def set_cache_dir(self, model_df: pd.DataFrame) -> None:
+        """
+        Set the cache directory in bucketfs of the specified model. Note that,
+        cache_dir class variable is used for testing purpose. This variable is
+        set to a local path only in unit tests.
+
+        :param model_df: The model dataframe to set the cache directory
+        """
+        model_name = model_df['model_name'].iloc[0]
+        bucketfs_conn_name = model_df['bucketfs_conn'].iloc[0]
+        sub_dir = model_df['sub_dir'].iloc[0]
+        bucketfs_location = bucketfs_operations.create_bucketfs_location(
+            self.exa.get_connection(bucketfs_conn_name))
+
+        model_path = bucketfs_operations.get_model_path(sub_dir, model_name)
+        self.cache_dir = bucketfs_operations.get_local_bucketfs_path(
+            bucketfs_location=bucketfs_location, model_path=str(model_path))
+
+    def load_models(self, model_name: str, **kwargs) -> None:
+        """
+        Load model and tokenizer model from the cached location in bucketfs
+
+        :param model_name: The model name to be loaded
+        """
+        self.last_loaded_model = self.base_model.from_pretrained(
+            model_name, cache_dir=self.cache_dir)
+        self.last_loaded_tokenizer = self.tokenizer.from_pretrained(
+            model_name, cache_dir=self.cache_dir)
+        self.last_created_pipeline = self.pipeline(
+            "token-classification",
+            model=self.last_loaded_model,
+            tokenizer=self.last_loaded_tokenizer,
+            framework="pt")
+
+        self.last_loaded_model = self.last_loaded_model.to(self.device)
+
+    def get_prediction(self, model_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Perform prediction of the given model and preparation of the prediction
+        results according to the format that the UDF can emit.
+
+        :param model_df: The dataframe to be predicted
+
+        :return: The dataframe where the model_df is formatted with the
+        prediction results
+        """
+        pred_df_list = self._predict_model(model_df)
+        pred_df = self._prepare_prediction_dataframe(model_df, pred_df_list)
+        return pred_df
+
+    def _predict_model(self, model_df: pd.DataFrame) -> List[pd.DataFrame]:
+        """
+        Predict the given text list using recently loaded models, return
+        probability scores, entities and associated words
+
+        :param model_df: The dataframe to be predicted
+
+        :return: List of dataframe includes prediction details
+        """
+
+        text_data = list(model_df['text_data'])
+        aggregation_strategy = model_df['aggregation_strategy'].iloc[0]
+        results = self.last_created_pipeline(
+            text_data, aggregation_strategy=aggregation_strategy)
```

Comment:
can you really give the aggregation_strategy into the call, the documentation looks like, as that you would need to specify the strategy at pipeline creation

Reply:
Yes, api document is not good enough I think.
Check tests in the without_db where different strategies are called.